homeassistant/components/goodwe/sensor.py

```diff
@@ -131,6 +133,10 @@ class GoodweSensorEntityDescription(SensorEntityDescription):
     key="_",
     state_class=SensorStateClass.MEASUREMENT,
 )
+ENUM_SENSOR = GoodweSensorEntityDescription(
```

Comment:
This doesn't really need its own here, it could be in the above list already... however, the bigger issue:

Nothing sets the possible options in this enum, which is required for sensors with the enum device class (preferably including translations for those).

../Frenck

Reply:
Yes, I did test it (am actually running it for weeks) and it does not throw any error and works properly.
The options are obviously not required (and technically inverter model/version specific, so not really suitable for explicit listing here). This is actually not a single sensor, there are 10 or so sensors which provide text values and share the same entity description.

Reply:
The options list is required when using the enum device class, otherwise the state machine will decline the state.

Reply:
Hmm, it indeed doesn't raise, which is an bug in our state implementation. Will address that, as this should not have worked.

../Frenck

Reply:
Hmm, as mentioned above, this is not a single sensor, there are cca 15-20 different sensors out there which are providing string values and had improperly configured unit as empty string so far. 
All these were working for a year or so, but since 2023.2 they started to emit ~10 of warnings every 30s.
If the ENUM might not be the best device class for them, how should they be registered ?
(They are diagnostic sensors providing text values and timestamp (without timezone).

Reply:
I am not in resolving it again, maybe I wasn’t clear enough by when using enum, the options are required.

I do not get why you mark it as resolved and u draft the PR without actually resolving anything.

If there are 15-20 different enum sensors, it would need 15-20 options mapped. 🤷‍♂️

Reply:
Sorry for resolving the conversation prematurely, it was mistake.
And yes, I understand the sensors were registered improperly from the beginning and the new checks/warning just finally identified that. And the goal is to fix that, the question is how.
They are not disabled, since they are actually used for monitoring their values, lots of users are used to them.
And even if disabled by default, it does not solve the question how to register them properly.
Some of them can be converted for real ENUM sensor, some not so easily or at all (considering serious differences between individual inverter models and their firmwares).
My question is, until I'll manage to convert (some) sensors to proper ENUM sensor, is there any other way how to provide sensors with string values without being ENUMs ?

Reply:
> My question is, until I'll manage to convert (some) sensors to proper ENUM sensor, is there any other way how to provide sensors with string values without being ENUMs ?

Yes, don't set a device class :)

Reply:
(facepallm) Obviously, sorry for stupid question.
I have uploaded change replacing the ENUMs.
If you do not object, I would suggest this as immediate fix to current state and I will submit subsequent PRs changing selected sensors to proper ENUMs inlc. localization.